releasenotes-builder/src/test/java/com/github/checkstyle/templates/TemplateProcessorTest.java

```diff
@@ -262,6 +262,34 @@ public void testGenerateCustomTemplate() throws Exception {
         assertFile("customTemplate.txt", MainProcess.GITHUB_FILENAME);
     }
 
+    @Test
+    public void testValidateNotesMinor() throws Exception {
+        final List<String> errors = MainProcess.runPostGenerationAndPublication(
+            createNotes(null, null, null, null, null), createBaseCliOptions("1.0")
+                .setValidateVersion(true).build());
+
+        Assert.assertEquals("one error", 1, errors.size());
+        Assert.assertEquals(errors.get(0),
+            "Validation of release number failed. Release number is minor(1.0), but "
+                + "release notes do not contain 'new' or 'breaking compatability' labels. Please "
+                + "correct release number by running https://github.com/checkstyle/checkstyle/"
+                + "actions/workflows/bump-version-and-update-milestone.yml");
+    }
+
+    @Test
+    public void testValidateNotesPatch() throws Exception {
+        final List<String> errors = MainProcess.runPostGenerationAndPublication(
+            createNotes(createAllNotes(), null, null, null, null), createBaseCliOptions("1.0.0")
+                .setValidateVersion(true).build());
+
+        Assert.assertEquals("one error", 1, errors.size());
```

Comment:
Please add tests for no violations for minor and patch

Reply:
Done.